src/Microsoft.ML.Tokenizers/Model/BPE.cs

```diff
@@ -444,36 +453,90 @@ internal List<Token> EncodeWithCache(string text)
                     return WordToTokens(ref word);
                 }
 
-                word = MergeWord(text.AsSpan());
-                Cache.Set(text, word);
+                word = MergeWord(text);
+                Cache.Set(text.ToString(), word);
```

Comment:
Do we know this cache helps beyond a microbenchmark on the same input each time?

Reply:
This model usually uses pre-tokenization which handle smaller words at a time. so cache will be helpful in such cases. We may consider adding a check to avoid caching for longer text (like more than 15 characters or so) to ensure it will be more effective.

Reply:
I have limited the max word length we store in the cache to 15 characters and ignore any longer text.